src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClient.kt

```diff
@@ -1,21 +1,14 @@
 package com.gamingbacklog.api.gamingbacklogapi.clients
 
-import com.fasterxml.jackson.databind.ObjectMapper
-import com.gamingbacklog.api.gamingbacklogapi.models.Game
+import com.gamingbacklog.api.gamingbacklogapi.clients.util.ExternalAPIClient
 import com.gamingbacklog.api.gamingbacklogapi.models.igdb.Credentials
 import com.gamingbacklog.api.gamingbacklogapi.models.igdb.IGDBGame
 import com.google.gson.Gson
 import org.springframework.beans.factory.annotation.Value
-import org.springframework.context.annotation.Bean
 import org.springframework.stereotype.Component
-import org.springframework.web.client.RestTemplate
-import java.net.URI
-import java.net.http.HttpClient
-import java.net.http.HttpRequest
-import java.net.http.HttpResponse
 
 @Component
-class IGDBClient {
+class IGDBClient(private val externalAPIClient: ExternalAPIClient) {
```

Comment:
Repeated since I think it got lost in the updates:
(Kotlin question) Is this showing ExternalAPICall as a class parameter? Do we want to extend the other class instead?

Reply:
(not sure how i missed so many comments, very weird)

This is basically the way Kotlin handles declaring spring-configured values. Instead of doing some sort of annotation like @Autowired we do it this way